### ISSUE TITLE
Address R CMD check NOTE

### ISSUE THREAD

**Issue by donyunardi** [insightsengineering/teal.modules.general#625]
NOTE from `devtools::check()`
```r
* checking package dependencies ... NOTE
Depends: includes the non-default packages:
  'ggmosaic', 'ggplot2', 'shiny', 'shinyTree', 'teal', 'teal.transform'
Adding so many packages to the search path is excessive and importing
selectively is preferable.
```

Let's fix this.

**Comment by pawelru** [comment]
related: https://github.com/insightsengineering/teal.modules.general/issues/352

**Comment by averissimo** [comment]
Note: moving any of them to `Imports` will create a different `NOTE` _(which I think is preferable)_

> ❯ checking package dependencies ... NOTE
>  Imports includes 21 non-default packages.
>  Importing from so many packages makes the package vulnerable to any of
>  them becoming unavailable.  Move as many as possible to Suggests and
>  use conditionally.

We could juggle packages to get around the NOTE in the short term:

- Import `%>%` from {dplyr} (which re-exports it, or even from [{teal}](https://github.com/insightsengineering/teal/blob/b1bbe12b2e6655db7daac1fc4fc9433a45ee33f5/NAMESPACE#L18) that also re-exports it)
- Remove {magrittr} from dependencies
  - Making room in `Imports` for 1 more package
- Move {teal.transform} from `Depends` to `Imports`
  - unless there's a reason why {teal.transform} absolutely needs to be in `Depends`
  - pinging @gogonzo as he added it on #618

**Comment by chlebowa** [comment]
I don't think this note needs fixing. All of the mentioned namespaces are used prolifically throughout tmg and their inclusion in `Depends` is warranted.

**Comment by gogonzo** [comment]
> Move {teal.transform} from Depends to Imports
unless there's a reason why {teal.transform} absolutely needs to be in Depends

Yes, it "has to" be in Depends. This follows the conclusion made years ago, that most of our users are novice and they might not understand namespaces at all. When using `teal` they want to have all the necessary functions ready to use. Consequence of this is following:
- NEST packages exposed to app developer are in depends
- examples in documentation (roxy and vignettes) are written without `::` prefixes for those functions.

**Comment by donyunardi** [comment]
> examples in documentation (roxy and vignettes) are written without :: prefixes for those functions.

Great point.
I'm proposing we handle this after the initial tmg release so we don't rush through the documentation. 
Any objection?

**Comment by chlebowa** [comment]
Solved in #663
